### DISCUSSION TITLE
`playlist` consumes files at maximum rate

### DISCUSSION THREAD

**Discussion by vitoyucepi** [savonet/liquidsoap#3744] [User Support]
**Describe the bug**
`playlist` will consume files at maximum speed when sending to `output.icecast` if liquidsoap sends `input.http` to `output.icecast` at the same time.

**To Reproduce**
```ruby
music = playlist("/tmp/test/music")

output.icecast(
  %mp3,
  music,
  host = "icecast",
  mount = "/a",
  fallible = true,
)

input_stream = input.http("http://icecast:8000/a")

output.icecast(
  %mp3,
  input_stream,
  host = "icecast",
  mount = "/b",
  fallible = true,
)
```

**Expected behavior**
`playlist` should consume files at a normal rate.

**Version details**

- OS: Debian in docker container
- Version: *2.2.4*

**Install method**
*savonet/liquidsoap:v2.2.4*

**Common issues**
#3626
#3627
https://github.com/savonet/liquidsoap/discussions/3727#discussioncomment-8427089

**Comment by toots** [comment]
Thank you for this report, will have a looks shortly.

**Comment by toots** [comment]
Wow that's an interesting bug!

I believe that this is behaving as intended. Let me explain:
* Icecast does not control the data flow, it simply relays the data at the rate it is given. The source client library is responsible for rate-control.
* When receiving an icecast stream, most likely than not, the stream will thus be sent rate-controled directly from the source-client.
* Liquidsoap has a notion of `self-sync` operator, which means that a given operator already has a built-in rate control. This is the case for `input.http` when connecting to an icecast server for the reason cited above.

However, with the given script, this means that the whole liquidsoap script then becomes rate-controlled by the rate at which `input.http` receives data.. which is the rate it is sent to by the first `output.icecast`.. which now isn't controlled at all by the machine's CPU like it is by default.

In other word, by sending and receiving to and from the same icecast server, the script rate-control is effectively disabled!

The solution is pretty simple, however: pass `self_sync=false` to `input.http` to prevent it from taking control of the data rate..

**Reply by vitoyucepi** [reply]
Hey @tdardly,
Take a look at the answer.
Basically, you should change your script like this.
```diff
-input_stream = clock(input.http("http://icecast:8000/a"))
+input_stream = input.http("http://icecast:8000/a", self_sync=false)
```

@toots,
In the original discussion at https://github.com/savonet/liquidsoap/discussions/3727#discussioncomment-8427089, I found that assigning a new `clock` to `input.http` also solves the problem.
